core/src/com/unciv/models/ruleset/unique/UniqueType.kt

```diff
@@ -451,6 +452,7 @@ enum class UniqueType(val text: String, vararg targets: UniqueTarget, val flags:
     HealOnlyByPillaging("Can only heal by pillaging", UniqueTarget.Unit, UniqueTarget.Global),
     HealsEvenAfterAction("Unit will heal every turn, even if it performs an action", UniqueTarget.Unit),
     HealAdjacentUnits("All adjacent units heal [amount] HP when healing", UniqueTarget.Unit),
+    HealInstantly("Doing so will consume this opportunity to choose a Promotion", UniqueTarget.Unit),
```

Comment:
Not sure I agree with this unique name and targets. It's use case is allowing modders to make promotions that can be taken multiple times, which isn't restricted at all to only healing instantly. Maybe `RepeatingPromotion` to make it similar to a repeating tech?
Additionally, it should only be allowed to put in Promotions, as it doesn't make sense for other Unit-like objects to have this unique.

Reply:
> it should only be allowed to put in Promotions, as it doesn't make sense for other Unit-like objects to have this unique.

Not sure how to declare where the promotion is allowed to be put. To use a different `UniqueTarget`? I still do not understand the difference between `UniqueTarget.Unit` and `UniqueTarget.Global`.

Reply:
The `UniqueTarget` indeed means where uniques can be put. For example, `UniqueTarget.Nation` means that it can only be placed on nations, `UniqueTarget.Promotion` only means promotions, etc. The full list can be viewed on `UniqueType.kt:12`. You'll also notice that some unique targets have things between paranthesis, e.g. `Nation(Global)`. This indicates inheritance, i.e., that anything that has `UniqueTarget.Global` can also be put on a Nation. For example, `[stats] from every [tileFilter]` has as `UniqueTarget.Global` (and `UniqueTarget.FollowerBelief`), so you could put those on nations as well.

In this case, a unique that can be put on `UniqueTarget.Global` can be put on, for example, policies, techs, nations, wonders, buildings, eras, resources and founder beliefs as these all (indirectly) inherit from `UniqueTarget.Global`. Uniques that can be put on `UniqueTarget.Unit` can be put on units, unit types and promotions, as these all inherit from `UniqueTarget.Unit`.

For example the unique `[amount]% Strength` has both `UniqueTarget.Unit` and `UniqueTarget.Global` as, when combined with conditionals, it should be allowed on either. `[-33]% Strength <vs cities>` is a common unique for mounted _units_, so `UniqueTarget.Unit`, and `[+25]% Strength <for [All] units> <for [30] turns>`, the unique for finishing the autocracy policy branch, should be put on a `UniqueTarget.Policy` -> `UniqueTarget.Global`. 

Does this help in understanding?

Reply:
> Does this help

_Now_ I'm confused. Just change the line to `UniqueTarget.Promotion` and move on...

Reply:
Thanks Xander, now I know more... it smells like even more unit tests are needed to cover that, e.g. `UniqueTarget.Promotion` is not applied on the `Nation`, etc... but they go out of this PR, definitely.